lib/rucio/daemons/rsedecommissioner/rse_decommissioner.py

```diff
@@ -0,0 +1,278 @@
+            kwargs={
+                'sleep_time': sleep_time,
+                'once': once
+            },
+        )
+    ]
+    [thread.start() for thread in threads]
+    # Interruptible joins require a timeout.
+    while any(thread.is_alive() for thread in threads):
+        [thread.join(timeout=3.14) for thread in threads]
+
+
+def stop(
+    signum: Optional[int] = None,
+    frame: Optional[FrameType] = None
+) -> None:
+    """
+    Graceful exit.
+    """
+    GRACEFUL_STOP.set()
+
+
+def decommission_rse(
+    rse: dict[str, Any],
+    profile: DecommissioningProfile,
+    *,
+    logger: Callable[..., None] = logging.log
+) -> None:
+    """RSE decommissioning template function.
+
+    RSE decommissioning proceeds in common steps of
+    - Profile initialization
+    - Discovery of rules to act (move / delete / etc.) on
+    - Actual handling of the rules
+    - Finalization
+    This function takes the functions corresponding to the four steps as its arguments and
+    executes them in order.
+
+    The handlers are given as a list of pairs of functions. The two functions both have a
+    signature
+    (rule, rse, *, logger=logging.log) -> bool
+    with the following actions:
+    - The first function is the "condition" against which each rule is tested. The returned
+      boolean indicates whether the rule satisfies it.
+    - The second function is the "(in)action" taken against the rule if the first function
+      returns True. The returned boolean indicates whether the limit checker should be
+      incremented after executing the action.
+
+    For each rule listed by the discoverer function, the conditions are applied in the order
+    given in the handlers list until one evaluates to True, at which point the corresponding
+    action function is executed. All remaining condition-action pairs are skipped for the rule.
+
+    :param rse: RSE table entry as a dictionary.
+    :param profile: Decommissioning profile.
+    :param logger: Logging function.
+    """
+    remove_max = {
+        'rules': config_get_int('rse-decommissioner', 'max_rules', default=-1),
+        'locks': config_get_int('rse-decommissioner', 'max_locks', default=-1)
+    }
+    if any(lim < 0 for lim in remove_max.values()):
+        raise RucioException('This daemon requires configuration parameters'
+                             ' rse-decommissioner/max_rules and rse-decommissioner/max_locks'
+                             ' set to non-negative integers.')
+
+    # Initialize the RSE for decommissioning.
+    profile.initialize(logger=logger)
+    # Counters for throttling.
+    num_untouched = 0
+    num_need_attention = 0
+    num_removed = {'rules': 0, 'locks': 0}
+
+    # Iterate over rules locking datasets / replicas at the RSE.
+    stop_reason = None
+    for rule in profile.discover(logger=logger):
+        outcome = profile.process(rule, logger=logger)
+
+        if outcome == HandlerOutcome.Untouched:
```

Comment:
Need all-caps.

```
lib/rucio/daemons/rsedecommissioner/rse_decommissioner.py:231:22: E1101: Class 'HandlerOutcome' has no 'Untouched' member (no-member)
lib/rucio/daemons/rsedecommissioner/rse_decommissioner.py:233:24: E1101: Class 'HandlerOutcome' has no 'Removed' member (no-member)
lib/rucio/daemons/rsedecommissioner/rse_decommissioner.py:252:24: E1101: Class 'HandlerOutcome' has no 'NeedAttention' member (no-member)
```

Reply:
I'm sorry for yet another trivial mistake. Don't know why this was missed.